guidance/models/transformers/_transformers.py

```diff
@@ -174,6 +77,204 @@ def __init__(
             transformers_tokenizer.eos_token_id,
         )
 
+    def _tokenizer(self, model: str, **kwargs) -> Union[
+        "transformers_package.PreTrainedTokenizer",
+        "transformers_package.PreTrainedTokenizerFast",
+    ]:
+        # make sure transformers is installed
+        if not has_transformers:
+            raise Exception("Please install transformers with `pip install transformers`")
+
+        try:
+            tokenizer = transformers_package.AutoTokenizer.from_pretrained(
+                model, use_fast=False, **kwargs
+            )
+        except ImportError as e:
+            # HuggingFace needs us to install something (sentencepiece, protobuf, etc. for some non-fast tokenizers)
+            raise RuntimeError(
+                f"Could not load tokenizer for model {model}. Please install the necessary dependencies for the tokenizer (see traceback for info)."
+            ) from e
```

Comment:
We'll now fail early rather than falling back to the fast tokenizer if the cause of the exception here is a missing dependency

Reply:
Interesting -- was guidance just eating a silent `ImportFailure` with e.g. attempting to load phi-3 without sentencepiece? I really like this change if we can catch those reasonably reliably.

Reply:
Yes, we were trying to use the `fast=False` tokenizer and had a catch-all except, which would fall back to the fast tokenizer.

Raising the `ImportError` seems like a good idea to me, as it gives the user something really actionable and specific to do to fix the issue. I'm still falling back to the fast tokenizer for any other exception, but maybe that's a bad idea if we know they are generally unreliable... 

Best case for me would be to only catch a specific exception class, e.g. `NotImplementedError` if that's what's thrown when there is no non-fast implementation or something... Unbounded `except` statements that silently fall back are scary (hence me adding a warning...)

Reply:
Now I dislike the runtime error. We should try to be as consistent as possible in this repo if we are raising for "you are missing a dependency" reasons.

Reply:
Agreed on dropping the runtime error